05_stringer/nicholeL/main_test.go

```diff
@@ -0,0 +1,38 @@
+package main
+
+import (
+	"bytes"
+
+	"github.com/stretchr/testify/assert"
+
+	"strconv"
+	"testing"
+)
+
+func TestMainOutput(t *testing.T) {
+	buf := bytes.Buffer{}
+	out = &buf
+	main()
+	expected := strconv.Quote("127.0.0.1")
+	actual := strconv.Quote(buf.String())
+	if expected != actual {
+		t.Errorf("Unexpected output in main()\nexpected: %q\nactual: %q", expected, actual)
+	}
+}
+
+func TestStringer(t *testing.T) {
+	testCases := []struct {
+		input    IPAddr
+		expected string
+	}{
+		{IPAddr{127, 0, 0, 1}, "127.0.0.1"},
+		{IPAddr{}, "0.0.0.0"},
+		{IPAddr{0, 0, 0, 0}, "0.0.0.0"},
+		{IPAddr{10, 74, 83, 7}, "10.74.83.7"},
+	}
+	for _, testCase := range testCases {
+		if !assert.Equal(t, testCase.input.String(), testCase.expected) {
```

Comment:
Would it be better to use the interface, like,
`actual := fmt.Printf("%v", testCase.input)`
instead of calling `testCase.input.String()`?

Reply:
Imo it's ok the way it is